test/sobol_validate.cpp

```diff
@@ -0,0 +1,371 @@
+  { 0.9257812504, 0.3554687502, 0.9492187504, 0.6523437503, 0.3476562502, 0.6132812503, 0.2851562501, 0.6523437503, 0.2382812501, 0.4570312502, 0.6523437503, 0.3945312502, 0.9335937504, 0.6757812503, 0.9726562505, 0.5039062502 },
+  { 0.4257812502, 0.8554687504, 0.4492187502, 0.1523437501, 0.8476562504, 0.1132812501, 0.7851562504, 0.1523437501, 0.7382812503, 0.9570312504, 0.1523437501, 0.8945312504, 0.4335937502, 0.1757812501, 0.4726562502, 0.0039062500 },
+  { 0.3007812501, 0.4804687502, 0.5742187503, 0.7773437504, 0.4726562502, 0.2382812501, 0.6601562503, 0.2773437501, 0.3632812502, 0.3320312502, 0.5273437502, 0.0195312500, 0.5585937503, 0.0507812500, 0.0976562500, 0.3789062502 },
+  { 0.8007812504, 0.9804687505, 0.0742187500, 0.2773437501, 0.9726562505, 0.7382812503, 0.1601562501, 0.7773437504, 0.8632812504, 0.8320312504, 0.0273437500, 0.5195312502, 0.0585937500, 0.5507812503, 0.5976562503, 0.8789062504 },
+  { 0.5507812503, 0.2304687501, 0.3242187502, 0.5273437502, 0.7226562503, 0.4882812502, 0.4101562502, 0.0273437500, 0.1132812501, 0.5820312503, 0.7773437504, 0.7695312504, 0.8085937504, 0.8007812504, 0.3476562502, 0.6289062503 },
+  { 0.0507812500, 0.7304687503, 0.8242187504, 0.0273437500, 0.2226562501, 0.9882812505, 0.9101562504, 0.5273437502, 0.6132812503, 0.0820312500, 0.2773437501, 0.2695312501, 0.3085937501, 0.3007812501, 0.8476562504, 0.1289062501 },
+  { 0.0351562500, 0.4648437502, 0.1523437501, 0.6679687503, 0.9570312504, 0.0664062500, 0.3320312502, 0.3867187502, 0.0664062500, 0.7539062504, 0.0429687500, 0.5351562502, 0.7304687503, 0.8476562504, 0.3632812502, 0.2382812501 },
+  { 0.5351562502, 0.9648437504, 0.6523437503, 0.1679687501, 0.4570312502, 0.5664062503, 0.8320312504, 0.8867187504, 0.5664062503, 0.2539062501, 0.5429687503, 0.0351562500, 0.2304687501, 0.3476562502, 0.8632812504, 0.7382812503 },
+  { 0.7851562504, 0.2148437501, 0.9023437504, 0.9179687504, 0.2070312501, 0.3164062501, 0.5820312503, 0.1367187501, 0.3164062501, 0.0039062500, 0.2929687501, 0.2851562501, 0.9804687505, 0.0976562500, 0.1132812501, 0.9882812505 },
+  { 0.2851562501, 0.7148437503, 0.4023437502, 0.4179687502, 0.7070312503, 0.8164062504, 0.0820312500, 0.6367187503, 0.8164062504, 0.5039062502, 0.7929687504, 0.7851562504, 0.4804687502, 0.5976562503, 0.6132812503, 0.4882812502 }
+};
+
+
+QRNG_VALIDATION_TEST_FUNCTIONS(sobol)
+
+BOOST_AUTO_TEST_CASE( validate_sobol )
+{
+  test_sobol_values(sobol_02_100, 0);
+  test_sobol_values(sobol_07_100, 128);
+  test_sobol_values(sobol_16_100, 128);
+}
+
+BOOST_AUTO_TEST_CASE( validate_sobol_seed )
+{
+  test_sobol_seed(sobol_02_100, 0);
+  test_sobol_seed(sobol_07_100, 128);
+  test_sobol_seed(sobol_16_100, 128);
+}
+
+BOOST_AUTO_TEST_CASE( validate_sobol_discard )
+{
+  test_sobol_discard(sobol_02_100, 0);
+  test_sobol_discard(sobol_07_100, 128);
+  test_sobol_discard(sobol_16_100, 128);
+}
```

Comment:
Matching published data is good, but can we compute the L2 star discrepancy and show it satisfies published bounds?